packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,287 @@
+      >
+        <KButton
+          v-for="(item, index) in allowedTimePeriods"
+          :key="`time-${item.timeframeLength}-${index}`"
+          :is-rounded="false"
+          :class="{'selected-option': item.timeframeText === selectedTimeframe.timeframeText}"
+          :data-testid="'select-timeframe-' + item.timeframeLength"
+          appearance="outline"
+          size="medium"
+          @click="changeTimeframe(item)"
+        >
+          {{ item.timeframeText }}
+        </KButton>
+      </div>
+    </template>
+    <template #footer>
+      <div class="d-flex justify-content-end">
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="clearSelection()"
+        >
+          Clear
+        </KButton>
+        <KButton
+          :is-rounded="false"
+          size="medium"
+          appearance="btn-link"
+          @click="submitTimeFrame()"
+        >
+          Apply
+        </KButton>
+      </div>
+    </template>
+  </KPop>
+</template>
+
+<script>
+// import { DatePicker } from 'v-calendar'
+import KButton from '@kongponents/kbutton/KButton.vue'
+import KPop from '@kongponents/kpop/KPop.vue'
+import KSegmentedControl from '@kongponents/ksegmentedcontrol/KSegmentedControl.vue'
+import { roundToNearestMinutes, getUnixTime } from 'date-fns'
+
+const allowedTimePeriods = [
+  {
+    section: 'Last',
+    timeframeText: '15 minutes',
+    timeframeLength: 60 * 15
+  },
+  {
+    section: 'Last',
+    timeframeText: 'Hour',
+    timeframeLength: 60 * 60 * 1
+  },
+  {
+    section: 'Last',
+    timeframeText: '3 hours',
+    timeframeLength: 60 * 60 * 3
+  },
+  {
+    section: 'Last',
+    timeframeText: '6 hours',
+    timeframeLength: 60 * 60 * 6
+  },
+  {
+    section: 'Last',
+    timeframeText: '12 hours',
+    timeframeLength: 60 * 60 * 12
+  },
+  {
+    section: 'Last',
+    timeframeText: '24 hours',
+    timeframeLength: 60 * 60 * 24
+  },
+  {
+    section: 'Last',
+    timeframeText: '7 days',
+    timeframeLength: 60 * 60 * 24 * 7
+  },
+  {
+    section: 'Last',
+    timeframeText: '30 days',
+    timeframeLength: 60 * 60 * 24 * 30
+  },
+  {
+    section: 'Last',
+    timeframeText: '90 days',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Week',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Month',
+    timeframeLength: 60 * 60 * 24 * 90
+  },
+  {
+    section: 'Previous',
+    timeframeText: 'Quarter',
+    timeframeLength: 60 * 60 * 24 * 90
+  }
+]
+
+export default {
+  name: 'KTimePicker',
+  components: {
+    KButton,
+    KPop,
+    KSegmentedControl,
+    DatePicker: () => import('v-calendar/lib/components/date-picker.umd')
+  },
+  props: {
+    value: {
+      type: Object,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      allowedTimePeriods,
+      mode: this.mode || 'custom',
```

Comment:
What is `this.mode` referencing here? I don't think this is valid

Reply:
Cleaned this up in my latest commit.